table/tables/index.go

```diff
@@ -338,7 +373,16 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 	} else {
 		idxVal = make([]byte, 0)
 		if distinct {
-			idxVal = EncodeHandleInUniqueIndexValue(h.IntValue())
+			if h.IsInt() {
+				idxVal = EncodeHandleInUniqueIndexValue(h.IntValue())
+			} else {
+				if opt.Untouched {
```

Comment:
Is it duplicate with L387?

Reply:
No L379 append the tailLen, L387 appends the untouched flag.

Reply:
Please add a comment for this.

Reply:
Should move L380 ~ 384 before L377?

Reply:
> Should move L380 ~ 384 before L377?

No, if the handle is int, there would be no tailLen at the first byte.